apis/r/tests/testthat/test-SOMADataFrame.R

```diff
@@ -67,6 +67,25 @@ test_that("SOMADataFrame creation", {
   expect_true(tbl1$Equals(tbl0$Filter(tbl0$bar < 5)))
 })
 
+test_that("int64 values are stored correctly", {
+  uri <- withr::local_tempdir("soma-indexed-dataframe")
+  asch <- arrow::schema(
+    arrow::field("foo", arrow::int32(), nullable = FALSE),
+    arrow::field("bar", arrow::int64(), nullable = FALSE),
+  )
+
+  sidf <- SOMADataFrame$new(uri)
+  sidf$create(asch, index_column_names = "foo")
+  tbl0 <- arrow::arrow_table(foo = 1L:10L, bar = 1L:10L, schema = asch)
+
+  sidf$write(tbl0)
+  tbl1 <- sidf$read()
+  expect_true(tbl1$Equals(tbl0))
+
+  # verify int64_downcast option was restored
+  expect_true(is.null(getOption("arrow.int64_downcast")))
```

Comment:
What if it was set before? Not unlikely on CI, but you or I could have set it for your system. Then this test goes *boom*.

Reply:
Good point. Just updated in 27ed9a04